app/models/compute_resource.rb

```diff
@@ -77,6 +77,10 @@ def self.providers_requiring_url
     _("Libvirt, oVirt, OpenStack and Rackspace")
   end
 
+  def self.providers_url_mandatory
+    %w(libvirt ovirt openstack rackspace)
```

Comment:
Yeah, or that if we want to keep it simple.
How about adding `validates :url, :presence => true` to each Foreman::Model class that needs the validation? I think, that's even better than to filter in the superclass?